examples/rosettacode/vignere_cipher.nit

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: Vigenère cipher
+#
+# See: <http://rosettacode.org/wiki/Vigenère_cipher>
+
+module vignere_cipher
+
+fun encrypt(src:String, key:String):String do
+    var out = new Buffer
+    var j:Int = 0
+
+    for i in [0..src.length-1] do
+        var c:Char = src[i]
+
+        if c >= 'a' and c <= 'z' then
+            c = c.to_upper
+        else if c < 'A' or c > 'Z' then
+            continue
+        end
+
+        out.add(((c.ascii + key[j].ascii - 2*'A'.ascii) % 26 + 'A'.ascii).ascii)
+        j = (j + 1) % key.length
+    end
+
+    return out.write_to_string
```

Comment:
usually we use `to_s` to transform a buffer into a string. in buffer, `write_to_string` is just an alias of `to_s` if a remember correctly